tests/app/billing/test_billing.py

```diff
@@ -282,118 +284,90 @@ def test_create_free_sms_fragment_limit(client, sample_service):
-                           headers=[('Content-Type', 'application/json'), create_authorization_header()])
+def test_create_free_sms_fragment_limit_past_year(client, sample_service):
 
-    data_new = {'financial_year_start': 2016, 'free_sms_fragment_limit': 9999}
+    data = {'financial_year_start': 2016, 'free_sms_fragment_limit': 9999}
     response = client.post('service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
-                           data=json.dumps(data_new),
+                           data=json.dumps(data),
                            headers=[('Content-Type', 'application/json'), create_authorization_header()])
 
     response_get = client.get(
         'service/{}/billing/free-sms-fragment-limit?financial_year_start=2016'.format(sample_service.id),
         headers=[('Content-Type', 'application/json'), create_authorization_header()])
 
     json_resp = json.loads(response_get.get_data(as_text=True))
-
     assert response.status_code == 201
     assert response_get.status_code == 200
-    assert json_resp['data']['financial_year_start'] == 2016
-    assert json_resp['data']['free_sms_fragment_limit'] == 9999
-
-
-def test_get_free_sms_fragment_limit_year_return_correct_data(client, sample_service):
-    years = [2016, 2017, 2018]
-    limits = [1000, 2000, 3000]
-
-    for i in range(0, len(years)):
-        annual_billing = {'financial_year_start': years[i], 'free_sms_fragment_limit': limits[i]}
-        response = client.post('service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
-                               data=json.dumps(annual_billing),
-                               headers=[('Content-Type', 'application/json'), create_authorization_header()])
-
-    for i in range(0, len(years)):
-        response_get = client.get(
-            'service/{}/billing/free-sms-fragment-limit?financial_year_start={}'.format(sample_service.id, years[i]),
-            headers=[('Content-Type', 'application/json'), create_authorization_header()])
-        json_resp = json.loads(response_get.get_data(as_text=True))
-        assert response_get.status_code == 200
-        assert json_resp['data']['free_sms_fragment_limit'] == limits[i]
+    assert json_resp['financial_year_start'] == 2016
+    assert json_resp['free_sms_fragment_limit'] == 9999
 
 
-def test_get_free_sms_fragment_limit_for_all_years(client, sample_service):
-    years = [2016, 2017, 2018]
-    limits = [1000, 2000, 3000]
-
-    for i in range(0, len(years)):
-        annual_billing = {'financial_year_start': years[i], 'free_sms_fragment_limit': limits[i]}
-        response = client.post('service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
-                               data=json.dumps(annual_billing),
-                               headers=[('Content-Type', 'application/json'), create_authorization_header()])
+def test_update_free_sms_fragment_limit(client, sample_service):
```

Comment:
Not sure if this test will actually update an existing `free_sms_fragment_limit`, 
if it is set in the default then assert the expected value in the DB before doing the POST, then do a POST which updates that limit and check that it has been updated via a db session

Reply:
Agree. Changed.